### ISSUE TITLE
Datatable throws "TypeError: Cannot read property 'localeCompare' of null" on sort

### ISSUE THREAD

**Issue by dmig** [quasarframework/quasar#1001]
### Software version

Quasar: 0.14.5
OS: Ubuntu 17.04
Node: 8.4.0
NPM: 5.4.2
Browsers: Chrome 61.0.3163.100 (Official Build)

### What did you get as the error?

> ```
> [Vue warn]: Error in getter for watcher "rows": "TypeError: Cannot read property 'localeCompare' of null"
> 
> found in
> 
> ---> <QDataTable>
>        <Catalog> at /home/dmig/dom.life/src/components/Catalog.vue
>          <QLayout>
>            <Main> at /home/dmig/dom.life/src/App.vue
>              <Root>
> warn @ vue.runtime.esm.js?a427:430
> handleError @ vue.runtime.esm.js?a427:515
> get @ vue.runtime.esm.js?a427:2771
> run @ vue.runtime.esm.js?a427:2844
> flushSchedulerQueue @ vue.runtime.esm.js?a427:2611
> (anonymous) @ vue.runtime.esm.js?a427:651
> nextTickHandler @ vue.runtime.esm.js?a427:598
> Promise resolved (async)
> timerFunc @ vue.runtime.esm.js?a427:613
> queueNextTick @ vue.runtime.esm.js?a427:661
> queueWatcher @ vue.runtime.esm.js?a427:2698
> update @ vue.runtime.esm.js?a427:2834
> notify @ vue.runtime.esm.js?a427:728
> reactiveSetter @ vue.runtime.esm.js?a427:958
> setSortField @ quasar.esm.js?f2cf:6865
> boundFn @ vue.runtime.esm.js?a427:164
> invoker @ vue.runtime.esm.js?a427:1733
> Vue.$emit @ vue.runtime.esm.js?a427:2202
> sort @ quasar.esm.js?f2cf:7075
> boundFn @ vue.runtime.esm.js?a427:164
> click @ quasar.esm.js?f2cf:7045
> invoker @ vue.runtime.esm.js?a427:1733
>
> vue.runtime.esm.js?a427:519 TypeError: Cannot read property 'localeCompare' of null
>     at string (webpack-internal:///3:6957:38)
>     at eval (webpack-internal:///3:7015:48)
>     at InternalArray.sort (native)
>     at Array.sort (native)
>     at VueComponent.sort (webpack-internal:///3:7015:12)
>     at VueComponent.boundFn [as sort] (webpack-internal:///1:165:14)
>     at VueComponent.rows (webpack-internal:///3:7351:14)
>     at Watcher.get (webpack-internal:///1:2775:25)
>     at Watcher.evaluate (webpack-internal:///1:2875:21)
>     at VueComponent.computedGetter [as rows] (webpack-internal:///1:3129:17)
> ```
> 
### What steps did you take, to get the error?
Create Datatable with single column and fill it with data, containing `null` values.
```vue
<template>
    <q-data-table
      :data="myData"
      :config="config"
      :columns="columns" />
</template>

<script>
      // ....
      myData: [
       {column: "text"},
       {column: "Text"},
       {column: "no text"},
       {column: ""},
       {column: null},
      ],
      config: {
        rowHeight: '58px'
      },
      columns: [
        { label: 'column', field: 'column', width: '100px', sort: true, type: 'string' }
      }
      // ....
</script>
```
Click on column header.

**Comment by dmig** [comment]
Good workaround is to use Intl.Collator(), because it handles `null` properly
```javascript
let collator = new Intl.Collator()
function stringCompare(a, b) {
  return collator.compare(a, b)
}
```
instead of
```javascript
function stringCompare(a, b) {
  return a.localeCompare(b)
}
```

**Comment by rstoenescu** [comment]
This is no longer a problem for v0.15. Closing as it's already in v0.15.

**Comment by rstoenescu** [comment]
If anyone wants to contribute with a fix for v0.14 branch, please feel free to PR. I am currently overloaded with work for v0.15 and SSR and would appreciate any help.
